### ISSUE TITLE
Unbound Alias List Not Showing

### ISSUE THREAD

**Issue by cmaxwe** [opnsense/core#5752]
**Important notices**

Before you add a new report, we ask you kindly to acknowledge the following:

- [ X] I have read the contributing guide lines at https://github.com/opnsense/core/blob/master/CONTRIBUTING.md
- [ X] I am convinced that my issue is new after having checked both open and closed issues at https://github.com/opnsense/core/issues?q=is%3Aissue

**Describe the bug**

The alias list shows as empty (there are several). The aliases in the list still work though.

**To Reproduce**

1. Navigate to the Services: Unbound DNS: Overrides page
2. Look at the list of aliases
3. It is empty

I am able to delete one of my host overrides and the list of alias starts showing. So it is odd...when I add that host override back in then it hides the list once again.

**Expected behavior**

Should see the list of aliases.

**Describe alternatives you considered**

Not sure there are any alternatives. I can delete the host override that is causing the issue edit my aliases then add the host override back I guess but that doesn't seem like a good solution.

**Screenshots**

With Desktop Override (broken):
![image](https://user-images.githubusercontent.com/2604119/166106409-c4bd8666-0f33-42b7-971d-0cf2d37072f4.png)

Without desktop override entry (working):
![image](https://user-images.githubusercontent.com/2604119/166106433-12810e99-cd79-4c95-89de-760ab03f5294.png)

The innocent host override:
![image](https://user-images.githubusercontent.com/2604119/166106475-8da8ebad-4900-4199-a6cf-83a4ce1882c1.png)

**Relevant log files**

Nothing in the logs that seems relevant.

**Additional context**

Can't think of any further context.

**Environment**

Software version used and hardware type if relevant, e.g.:

OPNsense 22.1.6 (amd64, OpenSSL).
Intel(R) Core(TM) i5-4570 CPU
Few Intel NICs + a bxe based NIC

**Comment by AdSchellevis** [comment]
Might be the same issue as https://forum.opnsense.org/index.php?topic=28043.msg136169#msg136169, in which case  https://github.com/opnsense/core/commit/4ec9ea3d1aad7333ef34f2d95fcaf091d01484da should fix the issue. 

```
opnsense-patch 4ec9ea3d
```


If not, best check if the dashboard reports errors or the browser console shows any signs.

**Comment by cmaxwe** [comment]
Patch did not fix it.

Nothing in the browser console and nothing in the dashboard.

The only error I get in the logs is:
[57569cf0-687f-4da6-801f-2ccd8414933a] Script action stderr returned "b'pgrep: Cannot get process list (kvm_getprocs: No such process)'"

**Comment by cmaxwe** [comment]
searchHostAlias is returning an empty result

![image](https://user-images.githubusercontent.com/2604119/166109993-e9ac2120-d953-48c5-9602-da01b1911842.png)

**Comment by AdSchellevis** [comment]
and you are aware that the aliases grid only show the ones beloning to the selected override? I noticed your first screenshot doesn't select anything, the second does.

**Comment by cmaxwe** [comment]
> and you are aware that the aliases grid only show the ones beloning to the selected override? I noticed your first screenshot doesn't select anything, the second does.

No - mark it as working as expected.

That is a really bad UX decision/anti-pattern.

There is no clear linkage between the two ux components and the checkboxes work like a radio button in this case.

**Comment by AdSchellevis** [comment]
it's a pretty standard master/detail form, but you can always open a pull-request with suggestions for change.

**Comment by CRFSlick** [comment]
I was confused by this myself and agree with @cmaxwe that this change is pretty unintuitive. It would be beneficial to have some indication that an entry must be selected for it's respective aliases to be displayed.

**Comment by cmaxwe** [comment]
> I was confused by this myself and agree with @cmaxwe that this change is pretty unintuitive. It would be beneficial to have some indication that an entry must be selected for it's respective aliases to be displayed.

Even if the table below had a title like "Alias - <Selected Host Override>" where the Host Override is the name of the selected entry. I think that would make it a bit more clear - wouldn't completely fix it but it would help.

I might take a look for this in the code and make the change when I get a few minutes.

**Comment by AdSchellevis** [comment]
maybe it's as simple as https://github.com/opnsense/core/commit/00bd62fbdd9a665720b782e23e44eb469d52181f ?

**Comment by CRFSlick** [comment]
That works, but I'm not sure about bundling multiple possibilities into a single case like.

How about instead we don't auto-select an entry for the user? That way, the page can have a default message when no case is selected, such as:

`Please select an entry to view it's aliases`

Just a thought.

**Comment by AdSchellevis** [comment]
We can't easily distinct between no results or none selected, we're a bit bound to what the grid (bootgrid) offers here.

**Comment by CRFSlick** [comment]
Alright. In that case, your previously proposed solution would work.

**Comment by cbrueffer** [comment]
Just a "me too" that the UI could be clearer here; just updated a system to 22.1.9_1, noticed that aliases (overrides) were not shown but still working and only realized the need to select a host entry after reading this PR. That said I'm used to the old "complete overview" and the `No results found for selected host or none selected` text didn't exactly jump into my face.

**Comment by duracell** [comment]
This is really annoying. To find an alias I really have to click on every entry to then search in the aliases?
And there is no possibility to change it back to the old complete list :(

**Comment by sorenstoutner** [comment]
The current UI also confused me and it wasn't until I came here that I figured out how to make it work.  Creating a new alias and then having nothing show up in the list is not possibly the correct implementation.  I would recommend that that all aliases are displayed by default (the way it used to be many versions ago).  If the display list is limited to a single host because it is selected above, I would recommend that this be prominently displayed on the screen.  Something along the lines of "Only showing aliases for domain.com".

**Comment by pilnik255** [comment]
I was so baffled by this as well until I read here that I simply have to select an override host above to have its aliases displayed. I think the way this was in the past - where all aliases were simply listed without the need of selecting host - was a lot more intuitive.

**Comment by Hydranet** [comment]
Just came here to say that the answer in this issue helped solved my confusion to why the aliases weren't showing up.

**Comment by Fclem** [comment]
Just came here with the same issue, and tbh this is not intuitive.
Those check boxes in Host overrides, really should be radios buttons, and there should be a way to have none selected, so as to see all aliases.
I had to read through this thread to figure out what was the issue.
TL;DR; for future reference :
Only aliases of the check-box selected host override will show up at any given time.

**Comment by StoneMoe** [comment]
confused me too. just took hours to debugging it and finally find this issue.

put children entries into a separate table which implies user that they are not related. and the checkbox is not work as that way.

**Comment by decmade** [comment]
I'm trying to understand how this is closed. This UX is garbage. Change it back.

**Comment by alexis-via** [comment]
This is funny how a bad UX design can lead to hours and hours of headaches among so many users (including me) that need to search on the internet the reason for what seems to be a bug, then arrive on this Github issue... to discover that, what they thought was a bug, was in fact just a bad UX design :) Crazy !

Please, just display the full list of aliases.

**Comment by henri9813** [comment]
Hello,

This UI/UX "feature" lead me to:
- search 2 minutes if i'm idiot or not.
- Make me try some api requests
- Make a google search hopping someone have the issue
- Find an old forum post on 24.1 series ( i'm on 26.1 serie )
- Find this beautiful github issue.

Just aliases tabs should not be visible while we don't select an override. OR at least, permit to select ALL entries, or just, show all by default, i don't see the interest of not showing all by default.

Where is the code ? i'm hot to propose a feature in php :) )

**Comment by swhite2** [comment]
Hi @henri9813,

The UX for this was recently reworked to be a tree view instead, but it's still on development.

To try it out on 26.1.5:

```
# opnsense-patch aa9f8af dde6f27
```
